modules/services/espanso.nix

```diff
@@ -16,9 +16,25 @@ let
     versionAtLeast
     ;
 
+  inherit (pkgs.stdenv.hostPlatform) isDarwin isLinux;
+
   cfg = config.services.espanso;
   espansoVersion = cfg.package.version;
 
+  package-bin =
+    if isLinux && cfg.x11Support && cfg.waylandSupport then
+      pkgs.writeShellScriptBin "espanso" ''
+        if [ -n "$WAYLAND_DISPLAY" ]; then
+          ${lib.meta.getExe cfg.package-wayland} "$@"
+        else
+          ${lib.meta.getExe cfg.package} "$@"
+        fi
+      ''
```

Comment:
Why do we need to create a script to choose which binary to launch? Just set the package you want and just use it.

Reply:
what would be your suggestion, for espanso to seamlessly work on both x11 and wayland by default?
Say If I switch from kde plasma x11 to wayland by logging out and logging in.

Reply:
`espanso-wayland` doesn't run on X11?

Reply:
No

Reply:
> espanso-wayland doesn't run on X11?

The `espanso` crate's `wayland` feature activates different logic for reading keyboard input.

Reply:
Hmm.. just feels like a lot of module options for a niche case of someone not committing to wayland or x11... 
Could change the example to be the script here, really...

Reply:
It allows anyone to run espanso by default is how I would look at it, without doing any debugging on why the espanso service is not starting/failing.

Reply:
That's fair.. just wanted to challenge the ideas here because it feels very weird.